### PR TITLE
Bump the version of the xUnit-Performance package

### PR DESCRIPTION
**Pull request by jorive**
- In a previous PR this was reverted and broke the Profile On/Off perf runs

### CODE REVIEW

File: tests/src/Common/external/external.depproj [hunk 1]
```diff
@@ -21,21 +21,11 @@
...
-    <PackageReference Include="xunit.performance.execution">
-      <Version>1.0.0-beta-build0003</Version>
-    </PackageReference>
-    <PackageReference Include="xunit.performance.metrics">
-      <Version>1.0.0-beta-build0003</Version>
-    </PackageReference>
-    <PackageReference Include="Microsoft.Diagnostics.Tracing.TraceEvent">
-      <Version>1.0.3-alpha-experimental</Version>
-    </PackageReference>
+    <PackageReference Include="xunit.performance.api" Version="1.0.0-beta-build0004" />
```
Comment (DrewScoggins) [inline tests/src/Common/external/external.depproj]: nit:  Maybe we should match how the rest of the package references are made.  I don't care if they are all single line style or the old multi-line style.

**Comment by DrewScoggins** [comment]
Once the smoketest performance tests run, we need to check the artifacts and ensure that the Off configurations do not have ETL files or any of the PMC metrics

**Comment by jorive** [comment]
@dotnet/rap-team PTAL

**Comment by jorive** [comment]
This changes fixes #11729

**Comment by AndyAyersMS** [comment]
Do we need to update the reference in the top-level `dependencies.props` ?

Seems wrong that we have this information repeated in so many places in the tree (though admittedly it used to be worse).
```
    <StaticDependency Include="@(XunitPerformanceDependency)">
      <Version>1.0.0-alpha-build0040</Version>
    </StaticDependency>
```

File: tests/src/Common/external/external.depproj (hunk 1 above)
Comment (jorive) [inline tests/src/Common/external/external.depproj]: Sure, I can do that.

**Comment by jorive** [comment]
@AndyAyersMS I like the idea of cleaning the `dependencies.prop` file, which to be honest, I did not know it existed. It does look out of date too and it would be a good idea to avoid duplicating this information across different files (which is unmanageable).
Having said that, I would prefer deferring it for a future PR.
Thoughts?

**Comment by DrewScoggins** [comment]
@dotnet-bot help

**Comment by DrewScoggins** [comment]
test OSX10.12 x64 Checked Build and Test

**Comment by DrewScoggins** [comment]
LGTM

**Comment by AndyAyersMS** [comment]
Changes LGTM too, but please follow up with the investigation into `dependencies.props`.

**Pull request merged**